packages/nextjs/test/config/loaders.test.ts

```diff
@@ -162,25 +170,41 @@ describe('webpack loaders', () => {
         resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/pages/api/nested/testApiRoute.js',
         expectedWrappingTargetKind: 'api-route',
       },
+      {
+        resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/pages/api/nested/testApiRoute.custom.js',
+        expectedWrappingTargetKind: 'api-route',
+      },
+      {
+        resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/app/nested/route.ts',
+        expectedWrappingTargetKind: 'route-handler',
+      },
+      {
+        resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/app/nested/route.custom.ts',
+        expectedWrappingTargetKind: 'route-handler',
+      },
       {
         resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/app/page.js',
         expectedWrappingTargetKind: 'server-component',
       },
+      {
+        resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/app/page.custom.js',
+        expectedWrappingTargetKind: 'server-component',
+      },
       {
         resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/app/nested/page.js',
         expectedWrappingTargetKind: 'server-component',
       },
       {
-        resourcePath: '/Users/Maisey/projects/squirrelChasingSimulator/src/app/nested/page.ts', // ts is not a valid file ending for pages in the app dir
-        expectedWrappingTargetKind: undefined,
```

Comment:
I checked and this is actually not true. It's just very uncommon because usually you have jsx in your pages.